pennylane/__init__.py

```diff
@@ -169,6 +169,28 @@ def circuit():
            qml.CNOT(wires=['q12', -1] )
            ...
 
+    Most devices accept a ``shots`` argument which specifies how many circuit executions
+    are used to estimate stochastic return values. In particular, ``qml.sample()`` measurements
+    will return as many samples as specified in the shots argument.
+
+    .. code-block:: python
+
+        dev = qml.device('default.qubit', wires=1, shots=10)
+
+        @qml.qnode(dev)
+        def circuit(a):
+          qml.RX(a, wires=0)
+          return qml.sample(qml.PauliZ(wires=0))
+
+        print(circuit(0.8))  # [ 1  1  1 -1 -1  1  1  1  1  1] -> 10 samples are returned
+       ...
+
+    The shots argument can be changed on a per-call basis using the built-in ``shots`` keyword argument:
+
+    ..code-block:: python
+
+        print(circuit(0.8, shots=3))  # [ 1  1  1] -> default is overwritten for this call
+        print(circuit(0.8))  # [ 1  1  1 -1 -1  1  1  1  1  1] -> default is reinstated
```

Comment:
maybe we shouldn't use 'reinstated', it makes it sound like we _are_ mutating something. "without shots specified, the QNode returns to using the device default"?